src/mpid/ch4/src/ch4r_recv.h

```diff
@@ -104,6 +112,11 @@ static inline int MPIDIG_handle_unexp_mrecv(MPIR_Request * rreq)
         mpi_errno = MPIDIG_reply_ssend(rreq);
         MPIR_ERR_CHECK(mpi_errno);
     }
+
+    if (MPIDIG_REQUEST(rreq, req->status) & MPIDIG_REQ_GPU_IPC) {
+        mpi_errno = MPIDIG_send_gpu_ipc_ack(rreq);
+        MPIR_ERR_CHECK(mpi_errno);
+    }
```

Comment:
Could you explain what is this case -- `MPIDIG_handle_unexp_mrecv`? e.g. when this happens?